### PR TITLE
Fix git email address parking possibility

### PR DESCRIPTION
**Pull request by sdkalchofner**
Fix users being able to block a git_email by adding it and never confirming it, thus disallowing any other users to add it.

Now only allows adding if the user hasn't already added it or it's not confirmed yet.

Initially had a `alter table drop constraint` for the unique, but SQLite doesn't know `constraint` and only `drop index`.
Then SQLite automatically names the indexes (not very pretty), and the index to remove was `sqlite_autoindex_git_email_2`.

I thought, oh hey, not very nice, but oh well, add another `DROP index if exists sqlite_autoindex_git_email_2`, but that results in: `index associated with UNIQUE or PRIMARY KEY constraint cannot be dropped`

tl;dr: need to drop the unique index and have to delete the DB's, and start up again because of SQLite and this change being needed

### CODE REVIEW

**Comment by sdkalchofner** [comment]
As a user should theoretically be allowed to send another confirmation in case he hasn't received one. I implemented the remove git-email fn also to remove the entry in the user_emails_sent table. This seems to be the simplest solution to go around this behaviour of only allowing one email per type to get sent to a user.

If you have a different idea/approach, please let me know. But this should be fine at least for the go-live :)

**Review by andyundso — APPROVED** [review]

**Comment by sdkalchofner** [comment]
one has to locally execute the command: `alter table git_email drop constraint git_email_email_key;` in the DB shell to remove the constraint, or delete and setup the DB new
